src/test/java/org/cryptomator/cryptofs/DeleteNonEmptyCiphertextDirectoryIntegrationTest.java

```diff
@@ -168,7 +168,7 @@ private Path firstEmptyCiphertextDirectory() throws IOException {
 
 	private boolean isEmptyDirectory(Path path) {
 		try (Stream<Path> files = Files.list(path)) {
-			return files.count() == 0;
+			return files.filter(p -> p.getFileName().toString().equals(Constants.DIR_ID_FILE)).count() == 0;
```

Comment:
`filter()` keeps, it doesn't discard. In other words: You're counting the number of files named `Constants.DIR_ID_FILE` here. Please add a unit test 😉

Reply:
to make it more concise and easier to extend:

```suggestion
			Predicate<Path> ignoredFiles = p -> Constants.DIR_ID_FILE.equals(p.getFileName().toString);
			return files.filter(ignoredFiles.not()).count() == 0;
```

Reply:
Fixed in [1f72a65](https://github.com/cryptomator/cryptofs/pull/118/commits/1f72a65ce639e8fe1fbd6e428bd6fb81268c0a1c)